pkg/proxy/conntrack/conntrack_test.go

```diff
@@ -20,253 +20,84 @@ limitations under the License.
+	f.tableType = tableType
+	f.ipFamily = family
+	f.filters = make([]*conntrackFilter, 0, len(netlinkFilters))
+	for _, netlinkFilter := range netlinkFilters {
+		f.filters = append(f.filters, netlinkFilter.(*conntrackFilter))
 	}
+	return uint(len(f.filters)), nil
 }
 
-func TestClearEntriesForNAT(t *testing.T) {
-	testCases := []struct {
-		name   string
-		origin string
-		dest   string
-
-		expectCommand string
-	}{
-		{
-			name:   "IPv4",
-			origin: "1.2.3.4",
-			dest:   "10.20.30.40",
-
-			expectCommand: "conntrack -D --orig-dst 1.2.3.4 --dst-nat 10.20.30.40 -p udp",
-		},
-		{
-			name:   "IPv6",
-			origin: "fd00::600d:f00d",
-			dest:   "2001:db8::5",
+var _ netlinkHandler = (*fakeHandler)(nil)
 
-			expectCommand: "conntrack -D --orig-dst fd00::600d:f00d --dst-nat 2001:db8::5 -p udp -f ipv6",
-		},
-	}
+func TestConntracker_ClearEntries(t *testing.T) {
 
-	for _, tc := range testCases {
-		ct := makeCT(success)
-		err := ct.ClearEntriesForNAT(tc.origin, tc.dest, v1.ProtocolUDP)
-		if err != nil {
-			t.Errorf("%s/success: unexpected error: %v", tc.name, err)
-		}
-		execCommand := ct.getExecutedCommand()
-		if tc.expectCommand != execCommand {
-			t.Errorf("%s/success: Expect command: %s, but executed %s", tc.name, tc.expectCommand, execCommand)
-		}
-
-		ct = makeCT(nothingToDelete)
-		err = ct.ClearEntriesForNAT(tc.origin, tc.dest, v1.ProtocolUDP)
-		if err != nil {
-			t.Errorf("%s/nothing to delete: unexpected error: %v", tc.name, err)
-		}
-	}
-}
-
-func TestClearEntriesForPortNAT(t *testing.T) {
 	testCases := []struct {
-		name string
-		port int
-		dest string
-
-		expectCommand string
+		name     string
+		ipFamily uint8
+		filters  []netlink.CustomConntrackFilter
 	}{
 		{
-			name: "IPv4",
-			port: 30211,
-			dest: "1.2.3.4",
-
-			expectCommand: "conntrack -D -p udp --dport 30211 --dst-nat 1.2.3.4",
+			name:     "single IPv6 filter",
+			ipFamily: unix.AF_INET6,
+			filters: []netlink.CustomConntrackFilter{
+				&conntrackFilter{
+					protocol: 17,
+					original: &connectionTuple{dstPort: 8000},
+					reply:    &connectionTuple{srcIP: netutils.ParseIPSloppy("2001:db8:1::2")},
+				},
+			},
 		},
 		{
-			name: "IPv6",
-			port: 30212,
-			dest: "2600:5200::7800",
-
-			expectCommand: "conntrack -D -p udp --dport 30212 --dst-nat 2600:5200::7800 -f ipv6",
+			name:     "multiple IPv4 filters",
+			ipFamily: unix.AF_INET,
+			filters: []netlink.CustomConntrackFilter{
+				&conntrackFilter{
+					protocol: 6,
+					original: &connectionTuple{dstPort: 3000},
+				},
+				&conntrackFilter{
+					protocol: 17,
+					original: &connectionTuple{dstPort: 5000},
+					reply:    &connectionTuple{srcIP: netutils.ParseIPSloppy("10.244.0.3")},
+				},
+				&conntrackFilter{
+					protocol: 132,
+					original: &connectionTuple{dstIP: netutils.ParseIPSloppy("10.96.0.10")},
+					reply:    &connectionTuple{srcIP: netutils.ParseIPSloppy("10.244.0.3")},
```

Comment:
if this turns out to be wrong, this ttest shoudl fail https://github.com/kubernetes/kubernetes/pull/126847/files#r1742000458